src/FixedSizeArray.jl

```diff
@@ -312,8 +313,18 @@ end
 
 parent_type(::Type{<:FixedSizeArray{<:Any,<:Any,Mem}}) where {Mem} = Mem
 
+"""
+    parent(f::FixedSizeArray)
+
+Return the underlying parent object of the fixed-size array `f`.
+Use `parentindices(f)` to get the valid indices of the parent of `f`.
+"""
+Base.parent(f::FixedSizeArray) = getfield(f, :mem)
+
+Base.parentindices(f::FixedSizeArray) = axes(parent(f))
+
 underlying_storage(m) = m
-underlying_storage(f::FixedSizeArray) = f.mem
+underlying_storage(f::FixedSizeArray) = parent(f)
```

Comment:
In principle now we could even remove `underlying_storage` entirely and always use `parent`, however I'm not sure that's safe in the case where in `copyto!` below we are copying into an object which has a parent itself.  Maybe it is?